src/app/shared/text-dialog.component.ts

```diff
@@ -0,0 +1,33 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Injector, Output, Input } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { BaseComponent } from 'app/shared/base.component';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+/**
+ * A component to be shown in a dialog, allowing to enter a comment inside a textarea
+ */
+@Component({
+  selector: 'text-dialog',
+  templateUrl: 'text-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TextDialogComponent extends BaseComponent {
```

Comment:
I think this was used for asking an ad question?
An alternative is to use `NotificationService.confirm` passing a mock custom field, just with internal name, name, required and type (`text`). The user will see a textarea and will type in before confirming.